server/resp/src/main/java/org/infinispan/server/resp/commands/set/SSCAN.java

```diff
@@ -0,0 +1,65 @@
+package org.infinispan.server.resp.commands.set;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+import org.infinispan.container.entries.CacheEntry;
+import org.infinispan.multimap.impl.EmbeddedSetCache;
+import org.infinispan.multimap.impl.SetBucket;
+import org.infinispan.server.iteration.IterationInitializationContext;
+import org.infinispan.server.iteration.IterationManager;
+import org.infinispan.server.iteration.map.MapIterationInitializationContext;
+import org.infinispan.server.resp.Resp3Handler;
+import org.infinispan.server.resp.commands.iteration.BaseIterationCommand;
+
+/**
+ * `<code>HSCAN key cursor [MATCH pattern] [COUNT count]</code>` command.
+ * <p>
+ * Scan the key-value properties of the hash stored under <code>key</code>.
+ *
+ * @since 15.0
+ * @see <a href="https://redis.io/commands/hscan/">Redis Documentation</a>
+ * @author José Bolina
+ */
+public class SSCAN extends BaseIterationCommand {
+
+   public SSCAN() {
+      super(-3, 1, 1, 1);
+   }
+
+   @Override
+   protected IterationManager retrieveIterationManager(Resp3Handler handler) {
+      return handler.respServer().getDataStructureIterationManager();
+   }
+
+   @Override
+   protected CompletionStage<IterationInitializationContext> initializeIteration(Resp3Handler handler, List<byte[]> arguments) {
+      EmbeddedSetCache<byte[], byte[]> multimap = handler.getEmbeddedSetCache();
+      return multimap.get(arguments.get(0)).thenApply(entry -> {
+         if (entry == null) return null;
+         return MapIterationInitializationContext.withSource(setToMap(entry));
+      });
+   }
+
+   private Map<byte[],byte[]> setToMap(SetBucket<byte[]> entry) {
+      return entry.toList().stream().collect(Collectors.toMap(v -> v, v-> "0".getBytes()));
```

Comment:
We could use an empty array in the value. With this approach, I think even filtering would work.